CMakeLists.txt

```diff
@@ -183,6 +183,7 @@ target_link_libraries(vcpkglib
         fmt::fmt
     PRIVATE
         Threads::Threads
+        $<$<NOT:$<OR:$<BOOL:${MSVC}>,$<BOOL:${APPLE}>>>:atomic>
```

Comment:
I don't think this the right approach. AFAIU there is just one system which needs it, and you need to enable it just for this system (user variable, or actual test during configuration). 
(I think this should be a responsibility of the CMake toolchain, but probably this doesn't help you.)

Reply:
Linux machines have no problem accepting the additional specification. This seems like the most effective way to achieve the goal of ensuring the library is listed without necessarily writing a ton of additional script. Do you have any suggestions?

Reply:
https://github.com/facebook/folly/blob/530a7266f9f15eaad196ce1b291fbbfc8e578810/CMake/folly-deps.cmake#L184-L212

I think we should look at something like this inside an `if(NOT MSVC)`

Reply:
Is it the use of an expression or the actual logic of the expression that is cause of concern?  The actual logic can be changed to avoid any mention of msvc. 

Two alternatives:

1.a) instead of using an expression, would this work better?

```
if (UNIX AND NOT APPLE)
  SET(ADDITIONAL_ATOMIC atomic)
endif()

find_package(Threads REQUIRED)
target_link_libraries(vcpkglib
    PUBLIC
        fmt::fmt
    PRIVATE
        Threads::Threads
        ${ADDITIONAL_ATOMIC}
)
```
1.b) 
```
find_package(Threads REQUIRED)
target_link_libraries(vcpkglib
    PUBLIC
        fmt::fmt
    PRIVATE
        Threads::Threads
)

if (UNIX AND NOT APPLE)
  target_link_libraries(vcpkglib PRIVATE atomic)
endif()
```


2) or, continuing with the use of the expression...
```
find_package(Threads REQUIRED)
target_link_libraries(vcpkglib
    PUBLIC
        fmt::fmt
    PRIVATE
        Threads::Threads
        $<$<AND:$<BOOL:${UNIX}>,$<NOT:$<BOOL:${APPLE}>>>:atomic>
)
```

If the intent is to script up dependency checks as with the folly link, I'll defer to the broader group as that's another level of planning/testing.

Reply:
I'd personally prefer 1.b, since this is not an area where generator expressions are important (`UNIX AND NOT APPLE` will not change between builds)

Reply:
> If the intent is to script up dependency checks as with the folly link

Yes, this was my intent. I'll take a look at drafting this.

Reply:
Ok, I've pushed changes to perform the test. @mprather can you ensure this still works for the Pi?

Reply:
@ras0219-msft I tried the change but it does not isolate the linker issue.  The code snippet actually compiles (I also tried manually as well) and thus the variable is not set.

Reply:
Aha, ok. What about if you replace the test code with:
```cpp
#include <cstdint>
#include <atomic>
std::atomic<uintptr_t> x;
std::atomic<uintmax_t> y;
int main(int, char**) {
  return x + y;
}
```
(taken from https://reviews.llvm.org/D81190)

Reply:
@ras0219-msft Your last suggestion is good. If you want to simplify the test, the following also works:

```
#include <atomic>
std::atomic<uint64_t> y;
int main(int, char**) {
  return y++;
}
```